### PR TITLE
[Azure/azure-cli] [Service Connector] `--secret name=xx keyVaultSecretUri=xx`: Add new parameter key value pair to support password from KeyVault

### PR DESCRIPTION
**Pull request by xfz11**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
service connector

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py [hunk 1]
```diff
@@ -196,7 +196,9 @@ def get_auth_info_params(auth_type):
                 Usage: --secret name=XX secret=XX
 
                 name    : Required. Username or account name for secret auth.
-                secret  : Required. Password or account key for secret auth.
+                secret  : One of <secret, keyVaultSecretUri, keyVaultSecretName> is required. Password or account key for secret auth.
+                keyVaultSecretUri  : One of <secret, keyVaultSecretUri, keyVaultSecretName> is required. Keyvault secret uri for secret auth.
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py:200]: please use lowercase, my suggestion: `vault-uri`, `vault-name`

File: src/azure-cli/azure/cli/command_modules/serviceconnector/action.py [hunk 2]
```diff
@@ -32,11 +32,22 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
...
+                d['secret_info'] = {
+                    'secret_type': 'keyVaultSecretReference',
+                    'name': v[0]
+                }
             else:
-                raise ValidationError('Unsupported Key {} is provided for parameter secret_auth_info. '
-                                      'All possible keys are: name, secret'.format(k))
+                raise ValidationError('Unsupported Key {} is provided for parameter secret_auth_info. All possible '
+                                      'keys are: name, secret, keyVaultSecretUri, keyVaultSecretName'.format(k))
         if len(d) != 2:
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/action.py:48]: the validation is not enough, e.g, users may pass `--secret secret=xx vault-uri=xx`

**Review by houk-ms — APPROVED** [review]
please address the comments

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py (hunk 1 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py]: updated

File: src/azure-cli/azure/cli/command_modules/serviceconnector/action.py (hunk 2 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/action.py:48]: if input is `--secret secret=xx vault-uri=xx` then `len(d) != 2`

**Comment by xfz11** [comment]
@kairu-ms  please help to review and merge the PR

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**